Build Linux AArch64 wheels natively without QEMU

jakelishman opened the pull request:
GitHub Actions now provides Linux runner images on native arm64 hardware[^1], which means we should be able to build the wheels natively, without requiring QEMU.  This potentially paves the way (pending a more complete move of CI to GHA) for Linux on AArch64 to potentially gain Tier 1 platforma support.

[^1]: https://github.blog/changelog/2025-01-16-linux-arm64-hosted-runners-now-available-for-free-in-public-repositories-public-preview/

<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
-->

### Summary



### Details and comments

jakelishman committed:
Build Linux AArch64 wheels natively without QEMU

GitHub Actions now provides Linux runner images on native arm64
hardware[^1], which means we should be able to build the wheels natively,
without requiring QEMU.  This potentially paves the way (pending a more complete
move of CI to GHA) for Linux on AArch64 to potentially gain Tier 1
platforma support.

[^1]: https://github.blog/changelog/2025-01-16-linux-arm64-hosted-runners-now-available-for-free-in-public-repositories-public-preview/

mtreinish commented on .github/workflows/wheels-build.yml:
```diff
@@ -173,16 +173,13 @@ jobs:
       - uses: actions/setup-python@v5
         with:
           python-version: ${{ inputs.python-version }}
       - uses: dtolnay/rust-toolchain@stable
```
I can't comment on L186 below but I think we can remove that line and use the default test command. Tier 2 we define as running the tests during build/release but not on PR CI. We limited it to just `test.python.compiler.test_transpiler` in the interest of runtime because the emulation was so slow. But if we have a native host we should have the time budget to run the full test suite which is more in spirit of what we have defined as tier 2 support.

jakelishman replied:
Done in 2b394e4099, let's see what happens.

jakelishman committed:
Remove restriction on tests

We previously restricted the tests to a small subset of the suite
because of runtime concerns with QEMU.  Since we are now running
natively, this removes the restriction, bringing Linux AArch64 in line
with what we expect from a Tier 2 platform.

mtreinish commented on .github/workflows/wheels-build.yml:
```diff
@@ -173,20 +173,16 @@ jobs:
-          platforms: all
       - uses: pypa/cibuildwheel@v2.22.0
         env:
           CIBW_ARCHS_LINUX: aarch64
```
We probably can also remove this line since it's implicitly the host arch. We only needed this to enable the emulation when running from x86.

jakelishman replied:
The tests of Pythons 3.9 and 3.10 had passed, so I've pushed up cd85bf783d that removes this explicit selection.

jakelishman committed:
Remove explicit architecture selection

jakelishman commented:
It's all good - Linux AArch64 passed all tests.  After #13474, there's a high chance that we can promote Linux AArch64 to Tier 1 platform support.

mtreinish reviewed (approved):
This LGTM, I'm excited to potentially promote linux aarc64 to tier 1 in 2.0. I think when we do decided to bump it up to tier 1 we should just add this runner to the tier 1 job definition to enable pgo etc since the config is now essentially the same as the normal ubuntu latest job. But for the time being keeping it separate to not enable pgo makes sense to me.

jakelishman commented:
Yeah, I deliberately didn't move it have PGO in the build process here because I somewhat considered that a Tier 1 thing.
